File: client-scalaz/src/main/scala/org/openapitools/client/api/Leaderboard.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import Leaderboard._

case class Leaderboard (
  id: Option[Long],
rankType: Option[String],
/* One of GLOBAL, FRIENDS, LOCAL, SEARCH, CUSTOM */
  rankMode: Option[String],
sortType: Option[String],
limitation: Option[Integer],
title: Option[String],
description: Option[String],
iconAsset: Option[AssetResponse],
bannerAsset: Option[AssetResponse],
application: Option[ApplicationMiniResponse],
metaData: Option[String],
active: Option[Boolean])

object Leaderboard {
  import DateTimeCodecs._

  implicit val LeaderboardCodecJson: CodecJson[Leaderboard] = CodecJson.derive[Leaderboard]
  implicit val LeaderboardDecoder: EntityDecoder[Leaderboard] = jsonOf[Leaderboard]
  implicit val LeaderboardEncoder: EntityEncoder[Leaderboard] = jsonEncoderOf[Leaderboard]
}

File: client-clojure/src/sirqul_io_t_platform/specs/leaderboard.clj
(ns sirqul-io-t-platform.specs.leaderboard
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.asset-response :refer :all]
            [sirqul-io-t-platform.specs.asset-response :refer :all]
            [sirqul-io-t-platform.specs.application-mini-response :refer :all]
            )
  (:import (java.io File)))


(def leaderboard-data
  {
   (ds/opt :id) int?
   (ds/opt :rankType) string?
   (ds/opt :rankMode) string?
   (ds/opt :sortType) string?
   (ds/opt :limitation) int?
   (ds/opt :title) string?
   (ds/opt :description) string?
   (ds/opt :iconAsset) asset-response-spec
   (ds/opt :bannerAsset) asset-response-spec
   (ds/opt :application) application-mini-response-spec
   (ds/opt :metaData) string?
   (ds/opt :active) boolean?
   })

(def leaderboard-spec
  (ds/spec
    {:name ::leaderboard
     :spec leaderboard-data}))
